Mathlib/Analysis/Distribution/SchwartzSpace.lean

```diff
@@ -667,13 +667,76 @@ open MeasureTheory FiniteDimensional
 class _root_.MeasureTheory.Measure.HasTemperateGrowth (μ : Measure D) : Prop :=
   exists_integrable : ∃ (n : ℕ), Integrable (fun x ↦ (1 + ‖x‖) ^ (- (n : ℝ))) μ
 
+open Classical in
+/-- An integer exponent `l` such that `(1 + ‖x‖) ^ (-l)` is integrable if `μ` has
+temperate growth. -/
+def _root_.MeasureTheory.Measure.integrablePower (μ : Measure D) : ℕ :=
+  if h : μ.HasTemperateGrowth then h.exists_integrable.choose else 0
+
+lemma integrable_pow_neg_integrablePower
+    (μ : Measure D) [h : μ.HasTemperateGrowth] :
+    Integrable (fun x ↦ (1 + ‖x‖) ^ (- (μ.integrablePower : ℝ))) μ := by
+  simp [Measure.integrablePower, h]
+  exact h.exists_integrable.choose_spec
+
 instance _root_.MeasureTheory.Measure.IsFiniteMeasure.instHasTemperateGrowth {μ : Measure D}
     [h : IsFiniteMeasure μ] : μ.HasTemperateGrowth := ⟨⟨0, by simp⟩⟩
 
 instance _root_.MeasureTheory.Measure.IsAddHaarMeasure.instHasTemperateGrowth {μ : Measure D}
     [h : μ.IsAddHaarMeasure] : μ.HasTemperateGrowth :=
   ⟨⟨finrank ℝ D + 1, by apply integrable_one_add_norm; norm_num⟩⟩
 
+lemma pow_mul_le_of_le_of_pow_mul_le {C₁ C₂ : ℝ} {k l : ℕ} {x f : ℝ} (hx : 0 ≤ x) (hf : 0 ≤ f)
+    (h₁ : f ≤ C₁) (h₂ : x ^ (k + l) * f ≤ C₂) :
+    x ^ k * f ≤ 2 ^ l * (C₁ + C₂) * (1 + x) ^ (- (l : ℝ)) := by
+  have : 0 ≤ C₁ := le_trans (by positivity) h₁
+  have : 0 ≤ C₂ := le_trans (by positivity) h₂
+  have : 2 ^ l * (C₁ + C₂) * (1 + x) ^ (- (l : ℝ)) = ((1 + x) / 2) ^ (-(l:ℝ)) * (C₁ + C₂) := by
+    rw [Real.div_rpow (by linarith) zero_le_two]
+    simp [div_eq_inv_mul, ← Real.rpow_neg_one, ← Real.rpow_mul]
+    ring
+  rw [this]
+  rcases le_total x 1 with h'x|h'x
+  · gcongr
+    · apply (pow_le_one k hx h'x).trans
+      apply Real.one_le_rpow_of_pos_of_le_one_of_nonpos
+      · linarith
+      · linarith
+      · simp
+    · linarith
+  · calc
+    x ^ k * f = x ^ (-(l:ℝ)) * (x ^ (k + l) * f) := by
+      rw [← Real.rpow_natCast, ← Real.rpow_natCast, ← mul_assoc, ← Real.rpow_add (by linarith)]
+      simp
+    _ ≤ ((1 + x) / 2) ^ (-(l:ℝ)) * (C₁ + C₂) := by
+      apply mul_le_mul _ _ (by positivity) (by positivity)
+      · exact Real.rpow_le_rpow_of_nonpos (by linarith) (by linarith) (by simp)
+      · exact h₂.trans (by linarith)
+
+lemma integrable_of_le_of_pow_mul_le
+    {μ : Measure D} [μ.HasTemperateGrowth] {f : D → E} {C₁ C₂ : ℝ} {k : ℕ}
+    (hf : ∀ x, ‖f x‖ ≤ C₁) (h'f : ∀ x, ‖x‖ ^ (k + μ.integrablePower) * ‖f x‖ ≤ C₂)
```

Comment:
Given that the constants don't appear in the conclusion, isn't this a job for `Bornology.IsBounded`? For example:
```lean
open Bornology Set in
example {f : D → E} :
    IsBounded (range f) ↔ (∃ C₁, ∀ x, ‖f x‖ ≤ C₁) := by
  rw [isBounded_iff_forall_norm_le]
  exact exists_congr fun C ↦ by simp
```

Reply:
Although on second thought, perhaps it might be out of style with the rest of this file (where you need more general comparison results which cannot simply be stated in the language of `IsBounded`). I guess the root cause is that `Function.HasTemperateGrowth` is stated as a comparison of the size rather than a boundedness statement.

With that in mind, I'll leave it to your discretion.

Reply:
The real interesting statement is the next lemma, for explicit bounds on the integrals. The current lemma is just an auxiliary statement, to be able to apply the next one (as there are often integrability side conditions). So, I feel it's important that the two lemmas are phrased with exactly the same set of assumptions. Let me add a comment on this.